spec/unit/analytics/ga4_import/page_view_consolidator_spec.rb

```diff
@@ -0,0 +1,32 @@
+require "spec_helper"
+
+RSpec.describe Analytics::Ga4Import::PageViewConsolidator do
+  let(:ga_data) do
+    [
+      Analytics::Ga4Import::PageData.new("/example1", "Title", 10),
+      Analytics::Ga4Import::PageData.new("/example2", "Title ", 100),
+      Analytics::Ga4Import::PageData.new("/example2", "Title", 10),
+      Analytics::Ga4Import::PageData.new("/example3", "Title", 10),
+      Analytics::Ga4Import::PageData.new("/example3/blah", "Title", 10),
+      Analytics::Ga4Import::PageData.new("/example4/blah?query_params", "Title", 60),
+      Analytics::Ga4Import::PageData.new("/example4/blah", "Title", 40),
+      Analytics::Ga4Import::PageData.new("/example3/y/blah", "Smart Answer", 1000),
+      Analytics::Ga4Import::PageData.new("http://example.com/example99", "Title", 10),
+      Analytics::Ga4Import::PageData.new("/not-real-path", "Page not found - GOV.UK", 99),
+    ]
+  end
+
+  subject(:consolidator) { described_class.new(ga_data) }
+
+  describe "#consolidated_page_views" do
+    it "consolidates page views by normalised path and sorts by views in descending order" do
+      expect(consolidator.consolidated_page_views).to eq([
+        ["/example2", 110],
+        ["/example4/blah", 100],
+        ["/example1", 10],
+        ["/example3", 10],
+        ["/example3/blah", 10],
+      ])
+    end
```

Comment:
We should have more than one assertion here as it's not clear looking at the input how they all fit together. 

I imagine the assertions you want here are:

it outputs a list of page views and paths
it orders the list by page views
it doesn't include page data that is excluded
it combines page views for the same path
it combines page views for page data that normalises to the same path